assets/js/blocks/price-filter/block.js

```diff
@@ -73,29 +81,61 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 				setMaxPrice( prices[ 1 ] );
 			}
 		},
-		[ minConstraint, maxConstraint, minPrice, maxPrice ]
+		[ minPrice, maxPrice, setMinPrice, setMaxPrice ]
 	);
 
 	// Track price STATE changes - if state changes, update the query.
 	useEffect( () => {
 		if ( ! attributes.showFilterButton ) {
-			debouncedUpdateQuery();
+			debouncedUpdateQuery( minPrice, maxPrice );
 		}
-	}, [ minPrice, maxPrice, attributes.showFilterButton ] );
+	}, [
+		minPrice,
+		maxPrice,
+		attributes.showFilterButton,
+		debouncedUpdateQuery,
+	] );
 
-	// Track PRICE QUERY changes so the slider reflects current filters.
+	// Track price query/price constraint changes so the slider reflects current filters.
+	const previousMinPriceQuery = usePrevious( minPriceQuery );
+	const previousMaxPriceQuery = usePrevious( maxPriceQuery );
+	const previousMinConstraint = usePrevious( minConstraint );
+	const previousMaxConstraint = usePrevious( maxConstraint );
 	useEffect( () => {
-		if ( minPriceQuery !== minPrice ) {
+		if (
+			! Number.isFinite( minPrice ) ||
+			( minPriceQuery !== previousMinPriceQuery && // minPrice from query changed
+				minPriceQuery !== minPrice ) || // minPrice from query doesn't match the UI min price
+			( minConstraint !== previousMinConstraint && // minPrice from query changed
+				minConstraint !== minPrice ) // minPrice from query doesn't match the UI min price
+		) {
 			setMinPrice(
 				Number.isFinite( minPriceQuery ) ? minPriceQuery : minConstraint
 			);
 		}
-		if ( maxPriceQuery !== maxPrice ) {
+		if (
+			! Number.isFinite( maxPrice ) ||
+			( maxPriceQuery !== previousMaxPriceQuery && // maxPrice from query changed
+				maxPriceQuery !== maxPrice ) || // maxPrice from query doesn't match the UI max price
+			( maxConstraint !== previousMaxConstraint && // maxPrice from query changed
+				maxConstraint !== maxPrice ) // maxPrice from query doesn't match the UI max price
+		) {
 			setMaxPrice(
 				Number.isFinite( maxPriceQuery ) ? maxPriceQuery : maxConstraint
 			);
 		}
-	}, [ minPriceQuery, maxPriceQuery, minConstraint, maxConstraint ] );
+	}, [
+		minPrice,
+		maxPrice,
+		minPriceQuery,
+		maxPriceQuery,
+		minConstraint,
+		maxConstraint,
+		previousMinConstraint,
+		previousMaxConstraint,
+		previousMinPriceQuery,
+		previousMaxPriceQuery,
+	] );
```

Comment:
This is probably the most complex change in this PR, this effect was relying on missing deps to be run only when the min/max query price or the min/max constraint changed. However, in order to fix the linting warnings we needed to add several missing dependencies. That mean the effect run more often than before, that's why I needed to add extra checks to ensure the query price or the constraints had changed.